consensus/polybft/stake_manager.go

```diff
@@ -193,6 +175,33 @@ func (s *stakeManager) init(blockchain blockchainBackend) error {
 	return s.state.StakeStore.insertFullValidatorSet(validatorSet)
 }
 
+func (s *stakeManager) getOrInitValidatorSet() (validatorSetState, error) {
+	validatorSet, err := s.state.StakeStore.getFullValidatorSet()
+	if err != nil {
+		if !errors.Is(err, errNoFullValidatorSet) {
+			return validatorSetState{}, err
+		}
+
+		validators, err := s.polybftBackend.GetValidators(0, nil)
```

Comment:
I would return an error here if the validators set returned here is empty.

Reply:
I will add this check and error inside validator snapshot because that is the place for it

Reply:
https://github.com/0xPolygon/polygon-edge/pull/1935/commits/d9188bed0a013ac2caad778cb7703ab86988ea8e